src/vl-mapactions-draw-action.js

```diff
@@ -32,8 +32,20 @@ export class VlDrawAction extends VlMapAction {
     });
     const drawInteraction = new Draw(options);
     interactions.push(drawInteraction);
-    if (options.snapping) {
-      interactions.push(new VlSnapInteraction(options.snapping.layer || layer));
+    if (options.snapping !== undefined) {
+      if (typeof options.snapping === 'boolean') {
+        if (options.snapping) {
```

Comment:
Kan dit niet '=== true' worden?

Reply:
Nee want als het dan false is denk je dat je nog in het andere geval zou kunnen zitten waar een object verwacht wordt.

Reply:
Misschien is dit een leesbaarder compromis?

```JS
switch (typeof options.snapping) {
  case 'boolean':
    if (options.snapping) {
      interactions.push(new VlSnapInteraction({source: layer.getSource()}));
    }
    break;
  case 'object':
    if (options.snapping.layer) {
      interactions.push(new VlSnapInteraction(options.snapping.layer.getSource(), options.snapping));
    } else {
      interactions.push(new VlSnapInteraction(layer.getSource(), options.snapping));
    }
    break;
  default:
    break;
}
```

Hierdoor heb je minder geneste if statements waardoor het, maar dat is potentieel persoonlijk, iets leesbaarder wordt. Anders mag het weer resolved worden.